src/garage/_dtypes.py

```diff
@@ -0,0 +1,51 @@
+"""Data types for agent-based learning."""
+import collections
+
+
+class TrajectoryBatch(
+        collections.namedtuple('TrajectoryBatch', [
+            'observations',
+            'actions',
+            'rewards',
+            'terminals',
+            'env_infos',
+            'agent_infos',
+            'lengths',
+        ])):
+    """A tuple representing a batch of whole trajectories.
+
+    A `TrajectoryBatch` represents a batch of whole trajectories produced when
+    one or more agents interacts with one or more environments.
+
+    Note:
+        Throughout this documentaiton, we denote the trajectory index dimension
+        as `N`, the variable-length time dimension of each trajectory as `(T)`,
+        and the single-step shape of a multi-dimensional data point as `*dims`.
+        `N * (T)` represents flattening the `N` and `(T)` dimensions into a
+        single fixed-length batch dimension, which can be imagined as arranging
+        all `N` trajectories along a single time axis, from left to right.
+
+    Attributes:
+        observations (numpy.ndarray): A numpy array of shape `[N * (T), *dims]`
+            containing the (possibly multi-dimensional) observations for all
+            time steps in this batch.
+        actions (numpy.ndarray): A  numpy array of shape `[N * (T), *dims]`
+            containg the (possibly multi-dimensional) actions for all time
+            steps in this batch.
+        rewards (numpy.ndarray): A numpy array of shape `[N * (T)]` containing
+            the rewards for all time steps in this batch.
+        terminals (numpy.ndarray): A boolean numpy array of shape `[N * (T)]`
+            containing the termination signals for all time steps in this batch
+        env_infos (numpy.record): A structured numpy array containing
+            arbitrary environment state information, stored as numpy arrays of
+            shape `[N * (T)]` or `[N * (T), *dims]`.
+        agent_infos (numpy.record): A structured numpy array containing
+            arbitrary agent state information, stored as numpy arrays of shape
+            `[N * (T)]` or `[N * (T), *dims]`. For example, this may contain
+            the hidden states from an RNN policy.
+        lengths (numpy.ndarray): A uint numpy array of shape `[N * (T)]`
```

Comment:
I think `lengths` has shape `[N]` since each entry represents the length of each trajectory.

Reply:
good point, thanks